code/modules/mob/living/basic/festivus_pole.dm

```diff
@@ -42,6 +42,8 @@
 
 /mob/living/basic/festivus/Initialize(mapload)
 	. = ..()
+	var/datum/action/toggle_seethrough/seeThrough = new
+	seeThrough.Grant(src)
```

Comment:
1. Instead of having the action give the component, just give the component to the mob, and have the component create the action. Makes it a bit easier to make use of your component. 

```dm
/mob/whatever/Initialize(mapload)
    . = ..()
    AddComponent(/datum/component/seethrough_mob)
```
```dm
/datum/component/seethrough_mob/Initialize(...)
    . = ..()
    var/datum/action/toggle_seethrough/action= new(src)
    action.Grant(parent)
```